plover/command/set_config.py

```diff
@@ -0,0 +1,41 @@
+import ast
+
+COMMAND_DELIMITER = ','
+
+
+def plugin_error(e, msg):
+    """ Add an additional message to an exception before raising to the engine. """
+    e.args = (e.args[0] + "\nSET_CONFIG: " + msg,)
+    raise e
+
+
+def set_config(engine, cmdline):
+    """
+    Set a Plover config option to a given value upon executing a stroke pattern.
+
+    Example usage (enter these lines into dictionary):
+    "O*EP": "{PLOVER:SET_CONFIG:translation_frame_opacity,100}",
+    "TR*PB": "{PLOVER:SET_CONFIG:translation_frame_opacity,0}",
+    """
+    # Break down the command string into a key and a value. Throw an error if it isn't in the form of a 2-tuple.
+    try:
+        key_str, value_str = tuple(cmdline.split(COMMAND_DELIMITER, 1))
+    except ValueError as e:
+        plugin_error(e,"Bad command string. Command must be in the form {PLOVER:SET_CONFIG:option,value}")
+    # Look up the old value of the config setting in order to make sure it's a valid option
+    try:
+        old_value = engine[key_str]
+    except KeyError as e:
+        plugin_error(e,"%s is not a valid Plover config option." % key_str)
+    # Evaluate the value string to convert to a Python literal.
+    try:
+        value_obj = ast.literal_eval(value_str)
+    except SyntaxError as e:
+        plugin_error(e, "Error parsing value string \"%s\"" % value_str)
+    # If it is (or can be coerced to) the type of the old value, set the option to the new value.
+    try:
+        engine[key_str] = value_obj
+    except (NameError, TypeError) as e:
+        plugin_error(e,"Config value type is incompatible: \"%s\" != %s." % (value_str, type(old_value)))
+    except ValueError as e:
+        plugin_error(e,"Value \"%s\" is not allowed for config option %s." % (value_str, key_str))
```

Comment:
I found the control-flow with so many try-excepts hard to reason about, but overall I think it's okay if you don't wanna make any changes. Just for fun, I tried splitting out each except into its own function and then replacing some comments with function names:

```python
def get_config_key_value_strings():
    try:
        config_key, value_str = tuple(cmdline.split(COMMAND_DELIMITER, 1))
        return (config_key, value_str)
    except ValueError as e:
        raise_command_error(e, 'Bad command string. Command must be in the form {PLOVER:SET_CONFIG:option,value}')

def get_current_config_value(config_key):
    try:
        return engine[config_key]
    except KeyError as e:
        raise_command_error(e, '%s is not a valid Plover config option.' % config_key)
def convert_config_value_string_to_literal(value_str):
    try:
        return ast.literal_eval(value_str)
    except SyntaxError as e:
        raise_command_error(e, 'Error parsing value string "%s"' % value_str)

def set_config(config_key, config_value, current_value)
    # If it is (or can be coerced to) the type of the old value, set the option to the new value.
    try:
        engine[config_key] = config_value
    except (NameError, TypeError) as e:
        raise_command_error(e, 'Config value type is incompatible: "%s" != %s.' % (config_value, type(current_value)))
    except ValueError as e:
        raise_command_error(e, 'Value "%s" is not allowed for config option %s.' % (config_value, config_key))

config_key, value_str = get_config_key_value_strings()
current_config_value = get_current_config_value(config_key)
new_config_value = convert_config_value_string_to_literal(value_str)
set_config(config_key, new_config_value, current_config_value)
```

(note: didn't run this at all 😛 )